modin/pandas/dataframe.py

```diff
@@ -364,41 +364,23 @@ def apply(
         )
         if not isinstance(query_compiler, type(self._query_compiler)):
```

Comment:
When does this case happen?

Reply:
it never happens, removed it

Reply:
found out that there's actually a case: when you pass a function name that reduces a frame into a scalar (for example `.size()`) then the [`BasePandasDataset.apply()`](https://github.com/modin-project/modin/blob/6550939753c76e896ef2bfd65bb9468d6ad161d7/modin/pandas/base.py#L820) returns a scalar instead of a query compiler: [`BasePandasDataset.apply()`](https://github.com/modin-project/modin/blob/6550939753c76e896ef2bfd65bb9468d6ad161d7/modin/pandas/base.py#L847) -> [`BasePandasDataset._string_function()`](https://github.com/modin-project/modin/blob/6550939753c76e896ef2bfd65bb9468d6ad161d7/modin/pandas/base.py#L633) -> [actual DataFrame method that returns a scalar](https://github.com/modin-project/modin/blob/6550939753c76e896ef2bfd65bb9468d6ad161d7/modin/pandas/base.py#L652-L655)
```python
>>> import modin.pandas as pd
>>> pd.DataFrame({"a": [1, 2, 3]}).apply("size")
3
```
Reverted the removal change

Reply:
Ok, thanks!